cmd/reporting-operator/main.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/operate-first/curator-operator/internal"
+	"github.com/operate-first/curator-operator/pkg/operator"
+	"github.com/operate-first/curator-operator/pkg/signals"
+	"github.com/spf13/cobra"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+const (
+	defaultServerPort = ":8080"
+)
+
+var (
+	reportLog = ctrl.Log.WithName("report")
+)
+
+var (
+	cfg operator.Config
+
+	rootCmd = &cobra.Command{
+		Use:   "reporting-operator",
+		Short: "",
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return cmd.Help()
+		},
+	}
+
+	startCmd = &cobra.Command{
+		Use:   "start",
+		Short: "Starts the http server",
+		Long:  "",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := runReporting(context.Background(), reportLog)
+			if err != nil {
+				reportLog.Error(err, "error generating report")
+			}
+		},
+	}
+)
+
+func init() {
+	rootCmd.AddCommand(startCmd)
+	startCmd.Flags().StringVar(&cfg.Port, "port", defaultServerPort, "http server port")
+}
+
+func main() {
+	opts := zap.Options{
+		Development: true,
+	}
+	logger := zap.New(zap.UseFlagOptions(&opts))
+	ctrl.SetLogger(logger)
+
+	rootCmd.AddCommand(startCmd)
+	err := rootCmd.Execute()
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func runReporting(ctx context.Context, logger logr.Logger) error {
+	var httpServerPort = cfg.Port
+	internal.NewHTTPServer(ctx, httpServerPort, logger)
+
+	server := http.Server{
+		Addr: httpServerPort,
+	}
+
+	logger.Info("starting http server on port" + httpServerPort)
+
+	go startServer(&server, logger)
+
+	stopCh, closeCh := signals.CreateChannel()
+	defer closeCh()
+	log.Println("system call notified:", <-stopCh)
+
+	err := shutdownServer(ctx, &server, logger)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func startServer(server *http.Server, logger logr.Logger) {
+	if err := server.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
+		logger.Error(err, "failed to listen on http server")
+	}
+	logger.Info("http server stopped gracefully")
+}
+
+func shutdownServer(ctx context.Context, server *http.Server, logger logr.Logger) error {
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
```

Comment:
Let's use a package constant for the default context timeout value. Also: is 5s too short here?